addons/mail/static/src/utils/test_utils.js

```diff
@@ -25,11 +25,13 @@ import {
     mock,
 } from 'web.test_utils';
 import Widget from 'web.Widget';
+import { getFixture } from "@web/../tests/helpers/utils";
 import { createWebClient, getActionManagerServerData } from "@web/../tests/webclient/helpers";
 
 import LegacyRegistry from "web.Registry";
 
-const { Component, EventBus } = owl;
+const { App, Component, EventBus } = owl;
+const { afterNextRender } = App;
```

Comment:
Why is that imported from App and not owl? And why is App global?
We might have serious namespacing issues when we try to use Owl in livechat embed code. (and already with `owl` being global rather than a module)

Reply:
App comes from owl, it's not a global

Reply:
My bad I missed that part. But shouldn't `afterNextRender` come from `owl` too then? As all other imports have been changed to be top level it would seem logical.

Reply:
afterNextRender is not provided by owl, we've patched it onto the App in our compatibility code, it's an odoo-ism. Ideally, each App would have an afterNextRender method and tests would use that, but currently since there are a lot of apps because of the various compatibility layers, it's not feasible to convert the calling code to do that.

Reply:
So the plan would be to make `afterNextRender` an instance method of `app` at a later point? I'm not sure if we need the granularity ourselves as our goal would probably to have a single app anyway, but that makes sense for having the most flexibility in the lib.

But then if it's an Odoo thing for now, shouldn't it be in an Odoo imported file rather than being pushed into App (for no apparent reason)?

Reply:
The creation of this method requires patching App early in the loading cycle so that we can intercept all creations of new Apps to account for their schedulers. I don't think we want to have this in owl or web at all but I didn't want to have to make a brand new compatibility file in mail and add it to the test_assets bundle to do this so I put it in the existing patch on App that we have in web for other reasons.